ZuluContent/Items/Misc/DragonEgg.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Server.Mobiles;
+
+namespace Server.Items
+{
+
+    public class DragonEgg : Item
+    {
+
+
+        public override double DefaultWeight
+        {
+            get { return 0.02; }
+        }
+
+
+        [Constructible]
+        public DragonEgg() : this(1)
+        {
+        }
+
+
+        [Constructible]
+        public DragonEgg(int amountFrom, int amountTo) : this(Utility.RandomMinMax(amountFrom, amountTo))
+        {
+        }
+
+
+        [Constructible]
+        public DragonEgg(int amount) : base(0x1725)
+        {
+            Stackable = true;
+            Amount = amount;
+            Name = "DragonEgg";
+            Hue = 33;
+        }
+
+        [Constructible]
+        public DragonEgg(Serial serial) : base(serial)
+        {
+        }
+
+        public override void OnDoubleClick(Mobile from)
+        {
+            if (!Movable)
+                return;
+
+            if (!IsChildOf(from.Backpack))
+            {
+                from.SendLocalizedMessage(1042001); // That must be in your pack for you to use it.
+                return;
+            }
+
+            Consume(1);
+            from.SendMessage("The egg begins to move and ");
+            
+            var p = new Point3D(from);
```

Comment:
Use the `FindValidSpawnLocation` function from `SpellHelper` like this:

```csharp
var p = new Point3D(from);
if (SpellHelper.FindValidSpawnLocation(from.Map, ref p, true))
{
}

```

This will prevent players from spawning ostards in potentially invalid locations, e.g. if the region doesn't allow it